opampserver/pkg/server/server.go

```diff
@@ -84,7 +84,7 @@ func StartOpAmpServer(ctx context.Context, logger logr.Logger, mgr ctrl.Manager,
 		var serverToAgent *protobufs.ServerToAgent
 		connectionInfo, exists := connectionCache.GetConnection(instanceUid)
 		if !exists {
-			connectionInfo, serverToAgent, err = handlers.OnNewConnection(ctx, deviceId, &agentToServer)
+			connectionInfo, serverToAgent, err = handlers.OnNewConnection(ctx, deviceId, &agentToServer, kubeClient)
```

Comment:
Will it make sense to store it in the `ConnectionHandlers` instead of passing it as argument to the function?